packages/runtime/src/widgets/Tag.tsx

```diff
@@ -10,23 +18,28 @@ export interface TagProps {
   id?: string;
   label: Expression<string> | Expression<string[]>;
   styles?: {
-    backgroundColor: string;
-    borderRadius: string;
-    fontSize: string;
+    backgroundColor?: Expression<string>;
+    borderRadius?: string;
+    fontSize?: string;
   };
   icon?: IconProps;
 }
-
 export const Tag: React.FC<TagProps> = (props) => {
   const [text, setText] = useState(props.label);
   const { values } = useRegisterBindings({ ...props, text }, props.id, {
     setText,
   });
+  const screen = useScreenContext();
+  const scope = useCustomScope();
+  const backgroundColor = evaluate(
+    screen as ScreenContextDefinition,
+    props?.styles?.backgroundColor,
+    scope
+  );
```

Comment:
You can just pass the expression through useRegisterBindings, it does the same thing and memoizes the value for you. It's important so you don't evaluate on each render.